### DISCUSSION TITLE
Aggregation Temporality Per Metric

### DISCUSSION THREAD

**Discussion by vishramachandran** [open-telemetry/community#1008] [Community Discussions]
This question is related to configuring Aggregation Temporality at time of exporting metrics.

1. Is it possible for developer to configure OTel SDK to export aggregation temporality at a finer granularity - at the metric level ? Essentially, I want to be able to say "I want delta temporality for metric X, but cumulative temporality for metric Y". Is this possible? This is of interest especially with UpDownCounters. Cumulative value is preferred because fetching current (and final) counter value at query time is efficient. But for other technical reasons (for example, possibility of stream aggregation of time series) I want to be able to do delta temporality for other metrics.

2. Is it possible to configure exporters to emit both delta and cumulative, and not just one? 

I see that OTLP proto carries temporality at the sample/time series granularity. This is excellent. However, the configurability in exporters (if any) doesn't stand out.

Could anyone point me to java and/or golang SDKs to where this is configurable ?

**Comment by trask** [comment]
Hi @vishramachandran,

There is a [very recent change in the spec](https://github.com/open-telemetry/opentelemetry-specification/pull/2404) aligning exactly with what you're looking for, in particular:

> DELTA (where Counters, Async Counters and Histograms all use Delta, UpDownCounter and Async UpDownCounter uses Cumulative)
